kubectl-mtb segfaults if a given benchmark doesn't exist

```
$ kubectl-mtb get benchmark foo
ID    NAME   CATEGORY   TYPE   PROFILE LEVEL
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1cb7653]

goroutine 1 [running]:
sigs.k8s.io/multi-tenancy/benchmarks/kubectl-mtb/internal/kubectl-mtb.printBenchmarks()
	/Users/machaffe/personal/multi-tenancy/benchmarks/kubectl-mtb/internal/kubectl-mtb/get.go:64 +0x213
sigs.k8s.io/multi-tenancy/benchmarks/kubectl-mtb/internal/kubectl-mtb.glob..func2(0x29f6b40, {0xc00070f180, 0x2, 0x2})
	/Users/machaffe/personal/multi-tenancy/benchmarks/kubectl-mtb/internal/kubectl-mtb/get.go:43 +0x19
github.com/spf13/cobra.(*Command).execute(0x29f6b40, {0xc00070ee60, 0x2, 0x2})
	/Users/machaffe/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:846 +0x5f8
github.com/spf13/cobra.(*Command).ExecuteC(0xc0002cf080)
	/Users/machaffe/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x3ad
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/machaffe/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
sigs.k8s.io/multi-tenancy/benchmarks/kubectl-mtb/internal/kubectl-mtb.Execute()
	/Users/machaffe/personal/multi-tenancy/benchmarks/kubectl-mtb/internal/kubectl-mtb/root.go:51 +0x25
main.main()
	/Users/machaffe/personal/multi-tenancy/benchmarks/kubectl-mtb/cmd/kubectl-mtb/main.go:18 +0x17
```

I think it may be caused by `filterBenchmarks()`. Working on a fix now

--- a/benchmarks/kubectl-mtb/README.md
+++ b/benchmarks/kubectl-mtb/README.md
@@ -40,6 +40,12 @@ Run benchmarks:
 kubectl-mtb run benchmarks -n "namespace" --as "user"
 ```
 
+Run a specific benchmark by ID:
+
+```bash
+kubectl-mtb run benchmark MTB-PL2-CC-TI-1 -n "namespace" --as "user"
+```
+
 ## Complete example
 
 ### Create a namespace

--- a/benchmarks/kubectl-mtb/internal/kubectl-mtb/get.go
+++ b/benchmarks/kubectl-mtb/internal/kubectl-mtb/get.go
@@ -27,7 +27,7 @@ import (
 )
 
 var getCmd = &cobra.Command{
-	Use:   "get <resource>",
+	Use:   "get [benchmark|benchmarks] [<benchmark ID>]",
 	Short: "display one or many benchmarks.",
 
 	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
@@ -35,7 +35,9 @@ var getCmd = &cobra.Command{
 			return err
 		}
 
-		filterBenchmarks(cmd, args)
+		if err := filterBenchmarks(cmd, args); err != nil {
+			return err
+		}
 		return nil
 	},
 

--- a/benchmarks/kubectl-mtb/internal/kubectl-mtb/run.go
+++ b/benchmarks/kubectl-mtb/internal/kubectl-mtb/run.go
@@ -35,7 +35,7 @@ import (
 var benchmarkRunOptions = types.RunOptions{}
 
 var runCmd = &cobra.Command{
-	Use:   "run <resource>",
+	Use:   "run [benchmark|benchmarks] [<benchmark ID>]",
 	Short: "run one or more multi-tenancy benchmarks",
 
 	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
@@ -48,7 +48,9 @@ var runCmd = &cobra.Command{
 			return err
 		}
 
-		filterBenchmarks(cmd, args)
+		if err := filterBenchmarks(cmd, args); err != nil {
+			return err
+		}
 		return nil
 
 	},

--- a/benchmarks/kubectl-mtb/internal/kubectl-mtb/util.go
+++ b/benchmarks/kubectl-mtb/internal/kubectl-mtb/util.go
@@ -2,6 +2,7 @@ package kubectl
 
 import (
 	"fmt"
+	"reflect"
 
 	"github.com/spf13/cobra"
 	"sigs.k8s.io/multi-tenancy/benchmarks/kubectl-mtb/pkg/benchmark"
@@ -15,7 +16,7 @@ func getResource(args []string) (string, error) {
 
 	r := args[0]
 	if !supportedResourceNames.Has(r) {
-		return "", fmt.Errorf("Please specify a valid resource")
+		return "", fmt.Errorf("Please specify one of the following resources: %v", reflect.ValueOf(supportedResourceNames).MapKeys())
 	}
 
 	return r, nil
@@ -29,14 +30,18 @@ func getBenchmarkArg(args []string) string {
 	return args[1]
 }
 
-func filterBenchmarks(cmd *cobra.Command, args []string) {
+func filterBenchmarks(cmd *cobra.Command, args []string) error {
 	profileLevel, _ := cmd.Flags().GetInt("profile-level")
 
 	id := getBenchmarkArg(args)
 	if id != "" {
 		b := test.BenchmarkSuite.ID(id)
+		if b == nil {
+			return fmt.Errorf("unknown benchmark ID: %s", id)
+		}
 		benchmarks = []*benchmark.Benchmark{b}
 	} else {
 		benchmarks = test.BenchmarkSuite.ProfileLevel(profileLevel)
 	}
+	return nil
 }

